iroh-p2p/src/service.rs

```diff
@@ -48,11 +50,11 @@ pub struct Libp2pService {
     swarm: Swarm<NodeBehaviour>,
     net_receiver_in: Receiver<RpcMessage>,
     bitswap_response_channels: HashMap<Cid, Vec<OneShotSender<Block>>>,
-    kad_queries: AHashMap<QueryId, QueryChannel>,
+    kad_queries: AHashMap<Key, QueryChannel>,
```

Comment:
This won't be the right keying for other queries, so I suggest adding an `enum QueryKey` or sth like that